### DISCUSSION TITLE
Safari support

### DISCUSSION THREAD

**Discussion by esthersweon** [foxglove/community#175] [Q&A]
We should officially support Safari, at least the latest version (currently 15.1). This applies to both macOS and iOS/iPadOS.

Note that supporting Safari doesn't automatically mean Studio will work well on touch devices, our interface is currently heavily optimized for mouse interaction and we have no immediate plans to improve that.

This ticket can be closed when we remove the browser warning on Safari.

Blockers: 

- https://github.com/foxglove/studio/issues/1541
- https://github.com/foxglove/studio/issues/2505

**Comment by defunctzombie** [comment]
Any notion of what doesn't work today? Breaking this down into smaller "this doesn't work" parts would be the first step in determining the work involved in supporting. I think generally its going to be tricky with more advanced uses of browser tech that typically appears in Chrome (i.e. filesystem support, maybe some bigint things?). I'm not sure this is something we will prioritize short term unless its a few small tweaks.

**Comment by esthersweon** [comment]
@defunctzombie – can go through and find the "this doesn't work" parts once we have a production web build going (i.e. after login flow is added).

I agree we should prioritize the obvious fixes, and then let users tell us if they have workflows that require any larger tweaks.

From @jhurliman: "i think the only real thing holding us back from iOS right now is a lack of polyfill for dataview bigint getters/setters"

**Comment by jtbandes** [comment]
I started working on that in https://github.com/foxglove/studio/commit/76e51ededb9c0d929b3c9a32abe3c5f984d3ef6b but it fell by the wayside; could pick it up again

**Comment by defunctzombie** [comment]
https://caniuse.com/mdn-javascript_builtins_dataview_getbigint64
https://caniuse.com/mdn-api_window_showopenfilepicker

**Comment by jtbandes** [comment]
Closing this issue as it is too broad (and includes a questionable premise: "most users use iOS").

The latest version of Safari supports BigInt64Array & DataView.getBigInt64.

Opened https://github.com/foxglove/studio/issues/2505 for showOpenFilePicker.

**Comment by amacneil** [comment]
@jtbandes besides individual features that need to land, I think it's worth having some point in time where we remove the warning banner and claim safari is supported. Is that time now? Or should we keep this ticket open to track that?

We can edit conjecture out of the ticket description...

**Comment by jtbandes** [comment]
Not sure what else may be missing. Probably someone needs to do a runthrough of all the features and make sure things are generally working. I guess if we want to claim Safari/Firefox are supported we probably shouldn't throw up a "performance may be degraded" toast either...

**Comment by jtbandes** [comment]
What would you envision us doing to make sure we maintain compatibility in the future (since most of our dev work happens in Chrome)?

**Comment by amacneil** [comment]
> What would you envision us doing to make sure we maintain compatibility in the future (since most of our dev work happens in Chrome)?

This is probably the real question. We probably need a Chromatic like service that can handle cross-browser testing.

Sounds like Chromatic already supports FF: https://www.chromatic.com/docs/browsers

**Comment by rgov** [comment]
I think you are missing OffscreenCanvas, no? https://caniuse.com/offscreencanvas

It's coming... eventually https://bugs.webkit.org/show_bug.cgi?id=183720

Edit: Oh I see, that's been worked around in foxglove/studio#1541 and related commits.

**Comment by Jooolee** [comment]
3D in Safari error, in chrome is ok

**Comment by jtbandes** [comment]
Closing as  duplicate of https://github.com/orgs/foxglove/discussions/174
